crates/pallet-domains/src/staking_epoch.rs

```diff
@@ -369,15 +369,20 @@ pub(crate) fn mint_funds<T: Config>(
 }
 
 pub(crate) fn mint_into_treasury<T: Config>(amount: BalanceOf<T>) -> Option<()> {
-    let existing_funds = TreasuryFunds::<T>::take();
+    let existing_funds = AccumulatedTreasuryFunds::<T>::get();
     let total_funds = existing_funds.checked_add(&amount)?;
+    if total_funds.is_zero() {
+        return Some(());
+    }
+
     match T::Currency::can_deposit(&T::TreasuryAccount::get(), total_funds, Provenance::Minted) {
```

Comment:
Funny, `can_deposit` will return `DepositConsequence::Success` if `total_funds` is zero...